test/lib/config.test.js

```diff
@@ -0,0 +1,15 @@
+import { getEntryByDotString } from '../../lib/config';
+
+describe('config', () => {
+  test('Should grab object entry by dot.string', () => {
+    const object = {
+      foo: {
+        bar: {
+          baz: 'foo',
+        },
+      },
+    };
+    expect(getEntryByDotString(object, 'foo.bar.baz')).toEqual('foo');
+    expect(getEntryByDotString(object, 'bar')).toBeUndefined();
```

Comment:
I would definitely add an undefined dotted querystring as well, since that reflects the purpose of the function well. 🙂 

```suggestion
    expect(getEntryByDotString(object, 'bar')).toBeUndefined();
    expect(getEntryByDotString(object, 'baz.bar.foo')).toBeUndefined();
```

Reply:
Ah yeah! Added it as a quick example, but would've been better to make it a proper test from the start 👍